test/unit/gateways/airwallex_test.rb

```diff
@@ -179,15 +179,24 @@ def test_purchase_passes_request_id
     stub_comms do
       @gateway.purchase(@amount, @credit_card, @options.merge(request_id: request_id))
     end.check_request do |_endpoint, data, _headers|
-      assert_match(/request_/, data)
+      assert_match(request_id, data)
     end.respond_with(successful_purchase_response)
   end
 
   def test_purchase_passes_currency_code
     stub_comms do
       @gateway.purchase(@amount, @credit_card, @options.merge(currency: 'USD'))
     end.check_request do |_endpoint, data, _headers|
-      assert_match(/request_/, data)
+      # only look for currency code on the create_payment_intent request
+      assert_match(/USD/, data) unless data.include?('_purchase')
```

Comment:
This change is unrelated to the `descriptor` field. The previous assertion was not testing the currency code.